cmd/pebble/cmd_enter.go

```diff
@@ -0,0 +1,188 @@
+package main
+
+import (
+	"fmt"
+
+	"github.com/canonical/pebble/internal/logger"
+	"github.com/jessevdk/go-flags"
+)
+
+const shortEnterHelp = "Run pebble command in new pebble environment"
+const longEnterHelp = `
+The enter command behaves like the run command with the possibility of running
+a single pebble command after starting the service manager. It's intended to be
+used as an entrypoint in container images.
+
+When no subcommand is given, the behavior is identical to the run command.
+
+When a subcommand is given, the service manager is started in the background
+without starting "autostart" services (i.e. run --hold, unless --run flag is
+given), and the subcommand is run as a pebble client command that can
+communicate with the service manager instance in the background. The service
+manager is shut down after the commands finishes unless noted otherwise below.
+
+Only a subset of pebble commands is supported as a subcommand:
+
+  exec
+  help
+  ls
+  plan
+  services
+  start
+  version
+
+Furthermore, some commands have restrictions and idiosyncrasies:
+
+  exec, ls, plan, services, version
+    Service manager logging is disabled unless -v/--verbose flag is provided.
+
+  help, ls, plan, services, version
+    Service autostart (the --run flag) is prohobitied.
+
+  start, stop
+    Service manager continues running after the command finishes.
+
+  stop
+    Service autostart (the --run flag) is required.
+`
+
+type enterFlags int
+
+const (
+	enterFlagSilenceLogging enterFlags = 1 << iota
+	enterFlagNoServiceManager
+	enterFlagKeepServiceManager
+	enterFlagRequireServiceAutostart
+	enterFlagProhibitServiceAutostart
```

Comment:
I don't think the `Flag` word here adds anything to these names, and removing it would shorten them up a bit. For example, `enterFlagNoServiceManager` to `enterNoServiceManager`.